ntropy_sdk/ntropy_sdk.py

```diff
@@ -228,7 +228,54 @@ def wait_with_progress(self, poll_interval=None):
             raise NtropyError("Batch wait timeout")
 
 
+class BatchGroup(Batch):
+    def __init__(self, sdk, chunks, timeout = 10 * 60 * 60, poll_interval=10):
+        self._chunks = chunks
+        self._batches = []
+        self._results = []
+        self._pending_batches = []
+        self._sdk = sdk
+        self.timeout = time.time() + timeout
+        self.poll_interval = poll_interval
+        self.num_transactions = sum([len(chunk) for chunk in chunks])
+        self._finished_num_transactions = 0
+
+        self._enrich_batches()
+
+    def _enrich_batches(self):
+        for chunk in self._chunks:
+            self._batches.append(self._sdk._enrich_batch(chunk))
+            time.sleep(self._sdk.MAX_BATCH_SIZE / 1000)
+
+        self._pending_batches = self._batches.copy()
+
+    def poll(self):
+        if self._pending_batches:
+            pending_progress = 0
+            for batch in self._pending_batches:
+                resp, status = batch.poll()
+
+                if status == "finished":
+                    self._pending_batches.remove(batch)
+                    self._finished_num_transactions += batch.num_transactions
+                    self._results += resp.transactions
```

Comment:
Should we early return here if not  `_pending_batches`?

Reply:
Already do [above](https://github.com/ntropy-network/ntropy-sdk/pull/18/files#diff-79046e0edd8bad0e9844998f49da739cda16ccb9aa24a0f77a315f003182ce3dR253). Or you mean something else?